### PR TITLE
REF: tighten what we accept in TimedeltaIndex._simple_new

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/resample.py [hunk 1]
```diff
@@ -1787,8 +1789,10 @@ def asfreq(obj, freq, method=None, how=None, normalize=False, fill_value=None):
 
     elif len(obj.index) == 0:
         new_obj = obj.copy()
-        new_obj.index = obj.index._shallow_copy(freq=to_offset(freq))
 
+        idx = obj.index
+        new_index = type(idx)([], dtype=idx.dtype, name=idx.name, freq=freq)
```
Comment (jreback) [inline pandas/core/resample.py]: this looks a odd, isn't
```Index([], dtype=idx.name.....)``` more appropriate?

File: pandas/tests/resample/test_base.py [hunk 2]
```diff
@@ -106,7 +106,9 @@ def test_resample_empty_series(freq, empty_series, resample_method):
     if isinstance(s.index, PeriodIndex):
         expected.index = s.index.asfreq(freq=freq)
     else:
-        expected.index = s.index._shallow_copy(freq=freq)
+        idx = s.index
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: same comment as above, why not just collapse this so: ```s.index.asfreq(freq=freq)``` or ``s.copy()``

we don't need to have the shallow copy used like this in tests, nor work-arounds like this.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/resample/test_base.py]: implemented a helper _asfreq_compat for this, cleans up some code nicely

File: pandas/core/resample.py [hunk 3]
```diff
@@ -1797,3 +1795,28 @@ def asfreq(obj, freq, method=None, how=None, normalize=False, fill_value=None):
             new_obj.index = new_obj.index.normalize()
 
     return new_obj
+
+
+def _asfreq_compat(index, freq):
```
Comment (jreback) [inline pandas/core/resample.py]: why can we not just .asfreq(...)? this seems very odd to a) put here (its an Index method), and b) why doesn't this wrok?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/resample.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/resample.py]: there is no DatetimeIndex.asfreq or TimedeltaIndex.asfreq (and shouldnt be)

File: pandas/core/resample.py [hunk 4]
```diff
@@ -1797,3 +1795,28 @@ def asfreq(obj, freq, method=None, how=None, normalize=False, fill_value=None):
...
+    """
```
Comment (jreback) [inline pandas/core/resample.py]: should this be in the indexes sub-tree?

Comment (jbrockmendel) [reply pandas/core/resample.py]: if we start using it elsewhere then we can move it.  as long as its just here, id prefer to keep it private

**Comment by jbrockmendel** [comment]
gentle ping, id like to get this one in and move on to the PeriodArray and DTA counterparts

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**